### ISSUE TITLE
C210K: Run a 2k user scaletest on new system

### ISSUE THREAD

**Issue by stirby** [coder/internal#152]
Before increasing scale, we need to validate the new infrastructure on a cheaper 2k user scaletest. 

There are tight time constraints on producing a 10k user reference architecture. While we have a long-term vision of recurring scaletests to improve the stability and efficiency of our product, this effort is urgently seeking that documentation. So we should unblock the ability to scale, but not slow down timeline when discovering poorly optimized areas. 

In short, let's get Coder to 10k users (C210K), then create a routine for resolving scaling issues.
- If there are egregious issues that functionally prevent increasing scale, we should solve them.
- If there are limitations that will require us to modify our scaling formulae, let's document them.
- If we find leaks that may increase the cost but not block the larger scaletests, let's document them now and solve later.

**Comment by bpmct** [comment]
When we finish this one, we'll update https://coder.com/docs/admin/infrastructure/validated-architectures/2k-users
